### PR TITLE
Del project

### PR DESCRIPTION
**Pull request by jcvsilva**
New button in the project page, that will delete the project in the database.

### CODE REVIEW

File: src/views/Project.vue [hunk 1]
```diff
@@ -6,20 +6,32 @@
     <h2>Client: {{ project.client }}</h2>
     <h2>Stage: {{ project.stage }}</h2>
   </div>
+  <button class="delButton" @click="handleDelete">Delete this project</button>
+  <!-- <div class="delButton" @click="delProjectClick">Delete this project</div> -->
```
Comment (joaojesus81) [inline src/views/Project.vue:10]: Remove commented line

File: src/views/Project.vue [hunk 2]
```diff
@@ -6,20 +6,32 @@
...
 
 <script>
 import { useRoute } from "vue-router";
 import getSingleProject from "@/composables/getSingleProject.js";
+import removeProject from "@/composables/removeProject";
+import router from "@/router";
 
 export default {
-  setup() {
+  setup(props, { emit }) {
```
Comment (joaojesus81) [inline src/views/Project.vue:20]: Replace with:
`export default {
    props: [...],
    emits: ['reload'],`

File: src/views/Project.vue [hunk 3]
```diff
@@ -6,20 +6,32 @@
...
     loadSingleProject(route.params.project_number);
 
-    return { project };
+    const handleDelete = async () => {
+      await removeProject(route.params.project_number);
+      await router.push({
+        name: "Home",
+      });
+      await emit("reload");
+    };
```
Comment (joaojesus81) [inline src/views/Project.vue:32]: Add Spinner component

**Review by joaojesus81 — CHANGES_REQUESTED** [review]
Few changes to make. Please address

File: src/views/Project.vue (hunk 1 above)
Comment (jcvsilva) [reply src/views/Project.vue:10]: Done.

Comment (jcvsilva) [reply src/views/Project.vue]: Solved with commit: f91a260cf97436d69f11d6840cac0cff4c8ba3a6

File: src/views/Project.vue (hunk 2 above)
Comment (jcvsilva) [reply src/views/Project.vue:24]: Solved with the merge of develop, commit: a95e07a5838412a71c17ff4d85b331bba1a7f46c

File: src/views/Project.vue (hunk 3 above)
Comment (jcvsilva) [reply src/views/Project.vue:36]: Solved with the merge of develop, commit: a95e07a

**Review by joaojesus81 — APPROVED** [review]
Well done

**Pull request merged**